src/code_wars/sg1.py

```diff
@@ -0,0 +1,112 @@
+import math
+from queue import PriorityQueue
+
+
+class DialHomeDevice:
+    def __init__(self, galactic_map):
+        self.start, self.goal = None, None
+        self._galactic_gates = [
+            [self._create_gate(x, y, gate) for x, gate in enumerate(gates)]
+            for y, gates in enumerate(galactic_map.split("\n"))
+        ]
+        self._set_connections()
+
+    def _create_gate(self, x, y, gate_key):
+        if gate_key == "X":
+            return None
+        gate = Gate(x, y)
+        if gate_key == "S":
+            self.start = gate
+        elif gate_key == "G":
+            self.goal = gate
+        return gate
+
+    def _set_connections(self):
+        for gates in self._galactic_gates:
+            for gate in gates:
+                if gate:
+                    gate.connect(self._galactic_gates, self.goal)
+
+    def dial_home(self):
+        queue = PriorityQueue()
+        queue.put((0, self.start, []))
+        visited = {}
+        visited[self.start] = 0
+        shortest_route = None
+        shortest_distance = math.inf
+        while queue.qsize() > 0:
+            previous_distance, current, route = queue.get_nowait()
+            for gate in current.connections:
+                distance = previous_distance + math.dist(
+                    gate.location, current.location
+                )
+                if gate == self.goal:
+                    if distance < shortest_distance:
+                        shortest_route = route
+                        shortest_distance = distance
+                else:
```

Comment:
can you combine the two if statements and create if and?

Reply:
Turns out no.
`if gate == self.goal and distance < shortest_distance:`
then this should happen.
`shortest_route = route`
`shortest_distance = distance`
but 
`if gate == self.goal and distance >= shortest_distance:`
nothing else should happen, so they can't be combined.